### DISCUSSION TITLE
Implementation of IImageInfoDetector for original image format

### DISCUSSION THREAD

**Discussion by jz5** [SixLabors/ImageSharp#1914] [Show and tell] [answered]
I'm trying to implement `IImageFormat` to add my own image format. Can I implement the `IImageInfoDetector` to support the original image format?

The `Identify` method returns `IImageInfo`, so it requires a class that implements `IImageInfo` for the original image format. 

And `IImageInfo` has `ImageMetadata` type property, but `ImageMetadata` is sealed and internal constructor, so it cannot isntance `ImageMetadata` from other assembly.

```cs
public class MyImageInfo : IImageInfo
{
    public MyImageInfo(PixelTypeInfo pixelType, int width, int height, ImageMetadata metadata)
    {
        PixelType = pixelType;
        Width = width;
        Height = height;
        Metadata = metadata;
    }

    public PixelTypeInfo PixelType { get; }
    public int Width { get; }
    public int Height { get; }
    public ImageMetadata Metadata { get; }
}

public sealed class MyDecoder :IImageDecoder, IImageInfoDetector
{
    public IImageInfo Identify(Configuration configuration, Stream stream)
    {
        // Do something and create ImageInfo
        return new MyImageInfo(pixelType, width, height, metadata/*how to create?*/);
    }
}
```

**Comment by JimBobSquarePants** [comment, marked as answer]
>but ImageMetadata is sealed and internal constructor, so it cannot isntance ImageMetadata from other assembly.

Yep. That's a bug. If you fancy raising a PR to make the constructor public that would be great!
